src/node/stage/approved.rs

```diff
@@ -191,14 +211,21 @@ impl Approved {
         match self
             .section_keys_provider
             .key_share()
-            .and_then(|share| event.into_signed_network_event(share))
+            .and_then(|share| event.clone().into_signed_network_event(share))
         {
             Ok(event) => self.consensus_engine.vote_for(event),
-            Err(error) => log_or_panic!(
-                log::Level::Error,
-                "Failed to create NetworkEvent: {}",
-                error
-            ),
+            Err(error) => {
+                if AccumulatingEvent::ParsecPrune == event {
+                    debug!("ParsecPrune doesn't cause panic");
+                } else {
```

Comment:
Why is this special handling needed? Why would the `ParsecPrune` event fail? Is this because you might not have any key share at this point yet? If so, then I think it would be better (cleaner) to do the check before calling `vote_for_event(ParsecPrune)`.

Reply:
> Is this because you might not have any key share at this point yet?

Yes. will move the check to the calling place.